mne/report/tests/test_report.py

```diff
@@ -310,12 +310,13 @@ def test_render_non_fiff(tmp_path):
         fnames_out.append(fname_out)
 
     report = Report()
-    report.parse_folder(
-        data_path=tmp_path,
-        render_bem=False,
-        on_error="raise",
-        raw_butterfly=False,
-    )
+    with pytest.warns(RuntimeWarning, match="Channels contain different"):
```

Comment:
This BDF has channels:

```
['Fp1', ...,'EXG8', 'Status']
```

and prefiltering information:

```
0 HP: DC; LP: 417 Hz
...
71 HP: DC; LP: 417 Hz
72 No filtering
```

So, `Status` channel has different filter.

This channel is found as `stim channel` by `_check_stim_channel` function,
but the channel is included as same as other channels if it is read by `mne.io.read_raw_bdf` w/o any options.
Then, I remained as is.
(But the channel has no filtering, highpass/lowpass information is set as 0Hz-417Hz in info)

For EDF, such a `test.edf` has `EDF Annotations` channel but it is defined as `TAL` (annotation) channel and is removed in `_read_edf_header`.

In the previous code, all prefiltering information but last are used

```
prefiltering = [_edf_str(fid.read(80)).strip() for ch in channels][:-1]
```

and this PR changes like:

```
prefiltering = np.array([_edf_str(fid.read(80)).strip() for ch in channels])[
    sel
]
```

https://github.com/mne-tools/mne-python/pull/12441/files/fe10e185f28d04d575313e9a8f776a35dbad7a68#diff-1d508c4fe7f39a5d09e8cd44e2996647850b3398a3813e37e9e97654581b689cR939


As BDF's last channel is always `status`, then previous code cut the last.
https://www.biosemi.com/faq/file_format.htm

But I'm not sure if the last channel of EDF is always annotation or not.
https://www.edfplus.info/specs/edfplus.html#annotation

If it is confirmed that the last channel of EDF is annotation (or something which can be ignored), the code can be like:

```
prefiltering = np.array([_edf_str(fid.read(80)).strip() for ch in channels])[
    [x for x in sel if x != len(channels) - 1]
]
```

and then BDF file will not make warnings.

It may be better to check a stim channel by `_check_stim_channel`.
In this case, `stim_channel` must be passed from `_get_info` to `_read_header`.

Or we can parse all prefiltering information in `_read_header` and select by using "`sel` -`stim_channel_idxs`" in `_get_info`.

How do you think?

Reply:
If the last channel in a BDF file is *always* the status channel, then I think we should treat it separately, since by definition this is not a regular analog channel and thus prefilter information doesn't really apply.

Reply:
> It may be better to check a stim channel by _check_stim_channel.

I agree any channels considered to be STIM (either by a user input option or by the standard like BDF) should be excluded from the filter settings check.

Reply:
ok, I updated as:

* Remain all prefiltering information in `_read_edf_header`. They are also remained in raw._raw_extras
* To set highpass/lowpass in info in `_get_info`, select channels of sel but stim_channel_idxs

This will remove `Status` channel of BDF to set highpasss/lowpass correctly.

In addition, I noticed that `_read_gdf_header` remains full prefintering information and it should be managed in `_get_info`.
Now GDF also should have **correct** hihgpass/lowpass.

Note: I also noticed that `_read_gdf_header` remains full `physical_min`, `physical_max`, `digital_min`, `digital_max`.
In `_read_edf_header`, they are selected by `sel`.
Therefore, these values in `_get_info` (`physical_ranges = edf_info["physical_max"] - edf_info["physical_min"]` and `cals = edf_info["digital_max"] - edf_info["digital_min"]`) are not expected one.

But I've not used GDF and I'm not familiar with GDF, I leave them here.

Reply:
About tests, some failed with QT related (?) and it seems other PR tests shows same.

Reply:
I'll take a look soon. The Qt errors should be fixed if you rebase on main.